api/v1/policy_webhook.go

```diff
@@ -0,0 +1,63 @@
+// Copyright (c) 2021 Red Hat, Inc.
+// Copyright Contributors to the Open Cluster Management project
+
+package v1
+
+import (
+	"fmt"
+
+	"k8s.io/apimachinery/pkg/runtime"
+	ctrl "sigs.k8s.io/controller-runtime"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/webhook"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+)
+
+// log is for logging in this package.
+var policylog = logf.Log.WithName("policy-resource")
+
+func (r *Policy) SetupWebhookWithManager(mgr ctrl.Manager) error {
+	return ctrl.NewWebhookManagedBy(mgr).
+		For(r).
+		Complete()
+}
+
+//+kubebuilder:webhook:path=/validate-policy-open-cluster-management-io-v1-policy,mutating=false,failurePolicy=fail,sideEffects=None,groups=policy.open-cluster-management.io,resources=policies,verbs=create;update,versions=v1,name=policy.open-cluster-management.io.webhook,admissionReviewVersions=v1
+
+var _ webhook.Validator = &Policy{}
+
+// ValidateCreate implements webhook.Validator so a webhook will be registered for the type
+func (r *Policy) ValidateCreate() (admission.Warnings, error) {
+	policylog.Info("validate create", "name", r.Name)
+
+	return r.validate()
+}
+
+// ValidateUpdate implements webhook.Validator so a webhook will be registered for the type
+func (r *Policy) ValidateUpdate(_ runtime.Object) (admission.Warnings, error) {
+	policylog.Info("validate update", "name", r.Name)
+
+	return nil, nil
+}
+
+// ValidateDelete implements webhook.Validator so a webhook will be registered for the type
+func (r *Policy) ValidateDelete() (admission.Warnings, error) {
+	policylog.Info("validate delete", "name", r.Name)
+
+	return nil, nil
+}
+
+// validate admits a pod if a specific annotation exists.
+func (r *Policy) validate() (admission.Warnings, error) {
+	policylog.Info("Validating Policy by webhook")
+
+	// 1 character for "."
+	if len(r.GetNamespace())+len(r.GetName()) > 62 {
```

Comment:
I don't think Kubernetes supports non-ASCII characters for names, but just in case, you should be using `utf8.RuneCountInString` as opposed to `len` since I'm fairly certain `len` measures the bytes used by the string and not the character count. In most cases, ASCII characters would be used which only use one byte each.